tests/unit_tests/sparse_blas/source/sparse_gemm_buffer.cpp

```diff
@@ -0,0 +1,233 @@
+/*******************************************************************************
+* Copyright 2023 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions
+* and limitations under the License.
+*
+*
+* SPDX-License-Identifier: Apache-2.0
+*******************************************************************************/
+
+#include <algorithm>
+#include <complex>
+#include <cstdlib>
+#include <iostream>
+#include <limits>
+#include <vector>
+
+#if __has_include(<sycl/sycl.hpp>)
+#include <sycl/sycl.hpp>
+#else
+#include <CL/sycl.hpp>
+#endif
+
+#include "oneapi/mkl.hpp"
+#include "oneapi/mkl/detail/config.hpp"
+#include "sparse_reference.hpp"
+#include "test_common.hpp"
+#include "test_helper.hpp"
+
+#include <gtest/gtest.h>
+
+extern std::vector<sycl::device *> devices;
+
+namespace {
+
+template <typename fpType, typename intType>
+int test(sycl::device *dev, intType nrows_A, intType ncols_A, intType ncols_C,
+         double density_A_matrix, oneapi::mkl::index_base index,
+         oneapi::mkl::layout dense_matrix_layout, oneapi::mkl::transpose transpose_A,
+         oneapi::mkl::transpose transpose_B, fpType alpha, fpType beta, intType ldb, intType ldc) {
+    sycl::queue main_queue(*dev, exception_handler_t());
+
+    intType int_index = (index == oneapi::mkl::index_base::zero) ? 0 : 1;
+    std::size_t opa_nrows = static_cast<std::size_t>(
+        transpose_A == oneapi::mkl::transpose::nontrans ? nrows_A : ncols_A);
+    std::size_t opa_ncols = static_cast<std::size_t>(
+        transpose_A == oneapi::mkl::transpose::nontrans ? ncols_A : nrows_A);
+
+    // Input matrix in CSR format
+    std::vector<intType> ia_host, ja_host;
+    std::vector<fpType> a_host;
+    intType nnz = generate_random_matrix<fpType, intType>(nrows_A, ncols_A, density_A_matrix,
```

Comment:
Do I understand correctly that `std::rand()` is not seeded and that the same matrices would be generated every time the test suite is run?

Reply:
[`std::rand()`](https://en.cppreference.com/w/cpp/numeric/random/rand) is seeded with `std::srand(1)` by default so the inputs are always the same.

Reply:
Good, thanks!